rules/Php73/Rector/FuncCall/ArrayKeyFirstLastRector.php

```diff
@@ -137,7 +137,7 @@ private function processArrayKeyFirstLast(
                 continue;
             }
 
-            if ($this->hasPrevCallNext($stmtsAware, $key + 2, $totalKeys, $keyFuncCall)) {
+            if ($this->hasInternalPointerChangeNext($stmtsAware, $key + 1, $totalKeys, $keyFuncCall)) {
```

Comment:
old + 2 is the correct way afaik, as it check next after

Reply:
keep using + 2 since it validate after the next of next

```suggestion
            if ($this->hasInternalPointerChangeNext($stmtsAware, $key + 2, $totalKeys, $keyFuncCall)) {
```

Reply:
Thanks for feedback. I was trying to understand what exactly it does. 
+2 was causing the issues in the case when call to `key` was inside `while` loop. It was working fine for:
```
function skipExecutionHasNextNext(array $arr) {
    reset($arr);
    $key = key($arr);
    $value = current($arr);
    echo $key . ": " . $value . PHP_EOL;
    next($arr);
}
```
So as I understand on the function level there were only 2 stmts - `reset` & `while` loop. However after my changes it might also look for calls that are happening before `key`, but inside the while loop.

Reply:
I see, let's give it a try 👍